src/deluge/gui/views/automation_view.cpp

```diff
@@ -3404,6 +3411,27 @@ uint32_t AutomationView::getMiddlePosFromSquare(int32_t xDisplay, int32_t effect
 	return squareStart;
 }
 
+// calculates value of all nodes within a square for automation editor rendering
+// and for display of square value in pad selection mode
+int32_t AutomationView::getAverageSquareKnobPosition(ModelStackWithAutoParam* modelStack, int32_t xDisplay,
+                                                     int32_t effectiveLength, int32_t xScroll, int32_t xZoom) {
+	int32_t squareStart = getPosFromSquare(xDisplay, xScroll, xZoom);
+	int32_t squareWidth = getSquareWidth(xDisplay, effectiveLength, xScroll, xZoom);
+	int32_t numNodesWithinSquare = squareWidth / kParamNodeWidth;
+
+	int32_t totalKnobPos = 0;
+
+	for (int32_t i = 0; i < numNodesWithinSquare; i++) {
+		int32_t value = modelStack->autoParam->getValuePossiblyAtPos(squareStart + (i * kParamNodeWidth), modelStack);
+		totalKnobPos =
+		    totalKnobPos + modelStack->paramCollection->paramValueToKnobPos(value, modelStack) + kKnobPosOffset;
```

Comment:
```suggestion
		totalKnobPos +=  modelStack->paramCollection->paramValueToKnobPos(value, modelStack);
```

`kNobPosOffset` should be added after the averaging, to reduce the risk of overflow and save us an add operation per loop.